### PR TITLE
eos_snmp_server: Fix to change communities without an ACL attached

### LINKED ISSUE #334: eos_snmp_server module doesn't change communities without an ACL attached
##### SUMMARY

The `eos_snmp_server` module performs no action when the `config.communities` attribute is set, unless the list items provide an ACL attribute. This is not information required by EOS, and the documentation has no indication of any such requirement.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

eos_snmp_server

##### ANSIBLE VERSION

```paste below
ansible [core 2.12.3]
  python version = 3.10.2 (main, Jan 17 2022, 00:00:00) [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)]
  jinja version = 3.0.1
  libyaml = True
```

##### COLLECTION VERSION

```paste below
Collection Version
---------- -------
arista.eos 4.1.1  
ansible.netcommon 2.6.1
```

##### OS / ENVIRONMENT

Arista vEOS
Software image version: 4.23.5M-cloud

##### STEPS TO REPRODUCE

```yaml
- hosts: eos1
  gather_facts: no
  connection: ansible.netcommon.network_cli
  become: yes

  vars:
    ansible_network_os: arista.eos.eos

  tasks:
    - name: SNMP Configuration
      arista.eos.eos_snmp_server:
        config:
          communities:
            - name: myreadcommunity
              ro: yes
            - name: mywritecommunity
              rw: yes
        state: merged
      register: snmp_out

    - debug:
        var: snmp_out
```

##### EXPECTED RESULTS

Above playbook should push config lines like:
```
snmp-server community myreadcommunity ro
snmp-server community mywritecommunity rw
```
This is valid config when applied manually.

##### ACTUAL RESULTS

The above playbook produces no change:
```
PLAY [eos1] ******************************************************************************************************************************************

TASK [SNMP Configuration] ****************************************************************************************************************************
ok: [eos1]

TASK [debug] *****************************************************************************************************************************************
ok: [eos1] => {
    "snmp_out": {
        "before": {},
        "changed": false,
        "commands": [],
        "failed": false
    }
}
```
However, if the task is changed to include ACL information:
```yaml
[...]
      arista.eos.eos_snmp_server:
        config:
          communities:
            - name: myreadcommunity
              ro: yes
              acl_v4: test
            - name: mywritecommunity
              rw: yes
              acl_v4: test
        state: merged
[...]
```
Then it does produce a change:
```
PLAY [eos1] ******************************************************************************************************************************************

TASK [SNMP Configuration] ****************************************************************************************************************************
changed: [eos1]

TASK [debug] *****************************************************************************************************************************************
ok: [eos1] => {
    "snmp_out": {
        "after": {
            "communities": [
                {
                    "acl_v4": "test",
                    "name": "myreadcommunity",
                    "ro": true
                },
                {
                    "acl_v4": "test",
                    "name": "mywritecommunity",
                    "rw": true
                }
            ]
        },
        "before": {},
        "changed": true,
        "commands": [
            "snmp-server community myreadcommunity ro test",
            "snmp-server community mywritecommunity rw test"
        ],
        "failed": false
    }
}
```

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gomathiselvi@gmail.com>
Fixes #334 

##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: tests/unit/modules/network/eos/test_eos_snmp_server.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def test_eos_snmp_server_merged(self):
             "snmp-server host host01 informs version 3 priv user01 udp-port 23",
             "snmp-server host host02 traps version 2c user01 udp-port 23",
             "snmp-server host host01 informs version 3 priv user01",
-            "snmp-server community comm5 rw list4",
+            "snmp-server community comm6 rw",
```
Comment (NilashishC) [inline tests/unit/modules/network/eos/test_eos_snmp_server.py:154]: Can we keep a test to make sure both with and without acls work?

Comment (GomathiselviS) [reply tests/unit/modules/network/eos/test_eos_snmp_server.py:154]: @NilashishC This particular testcase has a community configuration with acl `snmp-server community comm01 view view3 ro ipv6 list1`

**Review by NilashishC — APPROVED** [review]

**Comment by GomathiselviS** [comment]
regate

**Comment by GomathiselviS** [comment]
recheck

**Pull request merged**